akka-remote/src/main/scala/akka/remote/artery/tcp/TcpFraming.scala

```diff
@@ -93,7 +94,7 @@ import akka.util.ByteString
       }
 
       private def createBuffer(bs: ByteString): EnvelopeBuffer = {
-        val buffer = ByteBuffer.wrap(bs.toArray)
+        val buffer = ByteBuffer.wrap(bs.toArrayUnsafe())
```

Comment:
I think this actually pretty much always lead to a copy anyway, because of the buffer `concat`s in `ByteStringParser` and `.take` going on before calling `createBuffer`. But can't hurt to avoid copy if possible.